lib/tilejson.js

```diff
@@ -211,8 +211,21 @@ TileJSON.prototype.get = function(url, callback) {
     var opts = {
         url: url,
         encoding: null,
+        //TODO: find a clean solution
+        //tell requestjs to *not* unzip, on the fly when fetching
+        //vetor tiles (*.pbf *.mvt), because all tile* modules
+        //expect gzipped buffers.
+        //let requestjs unzip all other requests, as gzipped responses
+        //break stuff further down the line.
+        gzip: !(url.indexOf('.pbf') > -1 || url.indexOf('.mvt') > -1),
         timeout: tilejson.timeout,
-        headers: {Connection: 'Keep-Alive'},
+        headers: {
+            Connection: 'Keep-Alive',
+            //always send gzip header, to enable 3rd party vector tile
+            //sources that gzip their response according to this header
+            //being present
```

Comment:
This is more to follow the HTTP specification than to meet the needs of third-party sources.